app/web/platform/nativeLink.ts

```diff
@@ -0,0 +1,57 @@
+import { useEffect, useState } from "react";
+
+export function getReactNativeWebView(): typeof window.ReactNativeWebView {
+  if (
+    typeof window !== "undefined" &&
+    window.ReactNativeWebView !== undefined
+  ) {
+    return window.ReactNativeWebView;
+  }
+}
+
+export function isNativeEmbed(): boolean {
+  return getReactNativeWebView() !== undefined;
+}
+
+export function getNativeData() {
+  const wv = getReactNativeWebView();
+  if (!wv) return;
+  if (wv.injectedObjectJson()) {
+    return JSON.parse(wv.injectedObjectJson());
+  }
+}
```

Comment:
Do y'all have a policy on abbreviations? I usually work places where we try to avoid them for readability but just a nit.

Reply:
fixed